cookbook/add-tabs.rst

```diff
@@ -0,0 +1,300 @@
+                                <param name="og_image" />
+                            </param>
+                        </params>
+                    </property>
+                </properties>
+            </section>
+        </properties>
+    </form>
+
+
+
+Registering your own admin class
+--------------------------------
+
+We'll create a class for example ``src/Admin/PageAdminExtension`` which extends ``Sulu\Bundle\AdminBundle\Admin``.
+
+In the admin we need to implement configureViews. Also for our example we use the `webspace_manager` in combination with `security_checker` to check if the logged in user has permission to edit pages.
+
+Also important, notice how we set the parent to be `PageAdmin::EDIT_FORM_VIEW`. This adds our view to the `PageAdmin::EDIT_FORM_VIEW` view.
+
+The setFormKey is a reference to the <key> in the created form from the above section.
+
+.. code-block:: php
+
+    class PageAdminExtension extends Admin
+    {
+        /**
+        * @var ViewBuilderFactoryInterface
+        */
+        private $viewBuilderFactory;
+
+        /**
+        * @var WebspaceManagerInterface
+        */
+        private $webspaceManager;
+
+        /**
+        * @var SecurityCheckerInterface
+        */
+        private $securityChecker;
+
+        public function __construct(
+            ViewBuilderFactoryInterface $viewBuilderFactory,
+            WebspaceManagerInterface $webspaceManager,
+            SecurityCheckerInterface $securityChecker
+        ) {
+            $this->viewBuilderFactory = $viewBuilderFactory;
+            $this->webspaceManager = $webspaceManager;
+            $this->securityChecker = $securityChecker;
+        }
+
+        public function configureViews(ViewCollection $viewCollection): void
+        {
+            $formToolbarActionsWithoutType = [
+                new ToolbarAction('sulu_admin.save_with_publishing'),
+            ];
+
+            $routerAttributesToFormRequest = ['parentId', 'webspace'];
+            $routerAttributesToFormMetdata = ['webspace'];
+
+            $previewCondition = 'nodeType == 1';
+
+            if ($this->hasSomeWebspacePermission()) {
+                $viewCollection->add(
+                    $this->viewBuilderFactory
+                        ->createPreviewFormViewBuilder('sulu_page.page_edit_form.socials', '/socials')
+                        ->disablePreviewWebspaceChooser()
+                        ->setResourceKey('pages')
+                        ->setFormKey('page_socials')
+                        ->setTabTitle('Socials')
+                        ->setTabPriority(511)
+                        ->addToolbarActions($formToolbarActionsWithoutType)
+                        ->addRouterAttributesToFormRequest($routerAttributesToFormRequest)
+                        ->setPreviewCondition($previewCondition)
+                        ->setTitleVisible(true)
+                        ->setTabOrder(1025)
```

Comment:
Actually we tend to use binary numbers here, so that you could add more stuff in between later. So in that case it would be `2048`.